pkg/services/serviceaccounts/manager/service.go

```diff
@@ -39,6 +39,27 @@ func ProvideServiceAccountsService(
 	}
 	serviceaccountsAPI := api.NewServiceAccountsAPI(s, ac, routeRegister)
 	serviceaccountsAPI.RegisterAPIEndpoints(cfg)
+
+	basicKeys := store.GetNonServiceAccountAPIKeys(context.Background())
+	if len(basicKeys) > 0 {
+		s.log.Warn("Upgrading API keys to service accounts", "numberKeys", len(basicKeys))
+
+		for _, key := range basicKeys {
+			sa, err := store.CreateServiceAccountForApikey(context.Background(), key.OrgId, key.Name, key.Role)
+			if err != nil {
+				s.log.Warn("Failed to create service account for API key", "err", err, "keyId", key.Id)
+				return nil, err
+			}
+
+			err = store.UpdateApikeyServiceAccount(context.Background(), key.Id, sa.Id)
+			if err != nil {
+				s.log.Warn("Failed to attach new service account to API key", "err", err, "keyId", key.Id, "newServiceAccountId", sa.Id)
+				return nil, err
+			}
+			s.log.Warn("Updated basic api key", "keyId", key.Id, "newServiceAccountId", sa.Id)
```

Comment:
Suggesting new log levels:

```suggestion
		s.log.Debug("Upgrading API keys to service accounts", "numberKeys", len(basicKeys))

		for _, key := range basicKeys {
			sa, err := store.CreateServiceAccountForApikey(context.Background(), key.OrgId, key.Name, key.Role)
			if err != nil {
				s.log.Error("Failed to create service account for API key", "err", err, "keyId", key.Id)
				return nil, err
			}

			err = store.UpdateApikeyServiceAccount(context.Background(), key.Id, sa.Id)
			if err != nil {
				s.log.Error("Failed to attach new service account to API key", "err", err, "keyId", key.Id, "newServiceAccountId", sa.Id)
				return nil, err
			}
			s.log.Debug("Updated basic api key", "keyId", key.Id, "newServiceAccountId", sa.Id)
```